packages/frontend/src/app/modules/users/components/login/login.component.ts

```diff
@@ -77,10 +77,11 @@ export class LoginComponent implements OnInit, OnDestroy {
     this.hidePassword = true;
     this.loading = false;
     this.submitted = false;
-    this.returnUrl =
-      this.route.snapshot.queryParams.returnUrl ||
-      this.currentLanguageService.routePrefix;
+
     this.routePrefix = this.currentLanguageService.routePrefix;
+
+    const queryparam = this.route.snapshot.queryParamMap;
+    this.returnUrl = queryparam.get("returnUrl") ?? undefined;
```

Comment:
```suggestion
    this.returnUrl = queryparam.get("returnUrl") ?? "";
```
D'ailleurs, s'il retrouve rien, get renvoie quoi ? Si c'est undefined, tu peux oublier mon commentaire d'avant et si c'est null, tu peux changer le type par `string | null` et si c'est string vide, tu peux retirer le `?? ""`